ceph/rados/core_workflows.py

```diff
@@ -2234,21 +2235,161 @@ def create_inconsistent_object(self, pool_name, object_name):
         for obj_key in key_list:
             log.info(f" Deleting the {obj_key} for the {object_name} object")
             obj_del_check = self.rm_object_key(primary_osd, pg_id, object_name, obj_key)
+            time.sleep(5)
+            new_key_list = self.get_object_key_list(primary_osd, pg_id, object_name)
+            if obj_key in new_key_list:
+                log.error(
+                    f"The key:{obj_key} from object:{object_name} could not be deleted. Fail"
+                )
+                raise Exception("Object key not deleted error")
+            log.debug(f"Successfully Deleted the {obj_key} from object: {object_name}")
             rm_key_count = rm_key_count + 1
             if rm_key_count == 3 or not obj_del_check:
                 break
         if not self.change_osd_state(action="start", target=primary_osd):
             log.error(f"Unable to start the OSD : {primary_osd}")
             raise Exception("Execution error")
         log.info(f"Performing the deep-scrub on the pg-{pg_id}")
-        self.run_deep_scrub(pgid=pg_id)
-        # sleeping for 10 seconds, waiting for scrubbing to be over
+        if not self.start_check_scrub_complete(pg_id=pg_id):
+            log.debug(f"deep-scrubbing could not be completed on PG : {pg_id}")
+            raise Exception("PG not deep-scrubbed error")
+
+        # sleeping for 10 seconds
         time.sleep(10)
         health_check = self.check_inconsistent_health(True)
         assert health_check
         log.info(f"The inconsistent object is created in the pg{pg_id}")
         return pg_id
 
+    def start_check_scrub_complete(self, pg_id):
```

Comment:
we could have single module with parameter passed as "deep-scrub" if we want to check scrub or deep-scrub. However not mandatory, keeping separate is fine as well.

Reply:
Since the params checked are different, I thought it would be good to have two methods instead of if-else statements. Having two methods would also satisfy SOLID principles. 

I am in favour of 2 separate methods itself. If needed, can be merged into 1.